src/asset-management/AaveManager.sol

```diff
@@ -43,7 +43,13 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
         require(aPoolAddressesProvider != address(0), "AM: PROVIDER_ADDRESS_ZERO");
         addressesProvider = IPoolAddressesProvider(aPoolAddressesProvider);
         pool = IPool(addressesProvider.getPool());
-        dataProvider = IAaveProtocolDataProvider(addressesProvider.getPoolDataProvider());
+        updateDataProvider();
+    }
+
+    function updateDataProvider() public onlyOwner {
+        address lNewDataProvider = addressesProvider.getPoolDataProvider();
+        require(lNewDataProvider != address(0), "AM: DATA_PROVIDER_ADDRESS_ZERO");
+        dataProvider = IAaveProtocolDataProvider(lNewDataProvider);
     }
```

Comment:
Interesting, did you find out that data provider can change over time?

Reply:
I understood that from aave's [docs](https://docs.aave.com/developers/core-contracts/pooladdressesprovider) that 

1. the `PoolAddressesProvider` will never change
2. that the admins can set the many of the addresses such as `ACLManager`, `PriceOracle` etc

It seems that I may have incorrectly assumed that the `Pool` address will also never change just by virtue that it is a proxy contract. They even mentioned that it's recommended to fetch the pool address each time you need to interact with it. But I think we don't have to do an external call every time. I'll add a function to update the Pool address as well.

Reply:
alright, this I added